examples/binaries/stack-allocations/src/lib.rs

```diff
@@ -0,0 +1,169 @@
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! This contract recursively calls payload stack allocated read or load,
+//! depends on actions vector, which is set in init.
+//! For each recursion step we call check_sum, which is sum of all payload bytes.
+//! Then reply summary check_sum back to source account.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(feature = "std")]
+mod code {
+    include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
+}
+
+#[cfg(feature = "std")]
+pub use code::WASM_BINARY_OPT as WASM_BINARY;
+
+extern crate alloc;
+use alloc::vec::Vec;
+use codec::{Decode, Encode, MaxEncodedLen};
+
+#[derive(Encode, Decode)]
+pub struct InitConfig {
+    actions: Vec<Action>,
+}
+
+#[derive(Encode, Decode, Clone)]
+pub enum Action {
+    Read,
+    Load,
+}
+
+const HANDLE_DATA_SIZE: usize = 0x100;
+
+#[derive(Encode, Decode)]
+pub struct HandleData {
+    data: [u8; HANDLE_DATA_SIZE],
+}
+
+#[derive(Encode, Decode, MaxEncodedLen)]
+pub struct ReplyData {
+    check_sum: u32,
+}
+
+#[cfg(not(feature = "std"))]
+mod wasm {
+    use crate::{Action, HandleData, ReplyData, Vec};
+    use gstd::msg;
+
+    struct State {
+        actions: Vec<Action>,
+    }
+
+    static mut STATE: State = State {
+        actions: Vec::new(),
+    };
+
+    fn do_actions(mut actions: Vec<Action>) -> u32 {
+        let check_sum = match actions.pop() {
+            Some(Action::Read) => msg::with_read_on_stack(|payload| {
+                payload
+                    .map(|payload| payload.iter().fold(0u32, |acc, x| acc + *x as u32))
+                    .expect("Failed to read payload")
+            }),
+            Some(Action::Load) => {
+                let HandleData { data } =
+                    msg::load_on_stack().expect("Failed to load handle config");
+                data.iter().fold(0, |acc, x| acc + *x as u32)
+            }
+            None => return 0,
+        };
+        check_sum + do_actions(actions)
+    }
+
+    #[no_mangle]
+    extern "C" fn handle() {
+        let check_sum = do_actions(unsafe { STATE.actions.clone() });
+        msg::reply_on_stack(ReplyData { check_sum }, 0).expect("Failed to reply");
+    }
+
+    #[no_mangle]
+    extern "C" fn init() {
+        unsafe {
+            STATE.actions = msg::load_on_stack().expect("Failed to load init config");
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{Action, HandleData, InitConfig, ReplyData, HANDLE_DATA_SIZE};
+    use codec::Decode;
+    use gtest::{Program, System};
+    use rand::{Rng, SeedableRng};
+    use static_assertions::const_assert;
+
+    #[test]
+    fn stress() {
+        use Action::*;
+
+        const MAX_ACTIONS_AMOUNT: usize = 1000;
+        const MAX_NUMBER: u8 = 255;
+
+        // Check that check sum is less than u32::MAX
+        const_assert!(
```

Comment:
no reason in const assertions inside function body

Reply:
??

Reply:
Replace with common assert or move above from function body

Reply:
it must be inside function body and that it